soql-server-pg/src/main/scala/com/socrata/pg/server/QueryServer.scala

```diff
@@ -278,12 +313,35 @@ class QueryServer(val dsInfo: DSInfo, val caseSensitivity: CaseSensitivity) exte
                     CJSONWriter.writeCJson(datasetInfo, qrySchema,
                       r, reqRowCount, r.rowCount, dataVersion, lastModified, obfuscateId)(resp)
                   }
+                case InfoSuccess(copyNumber, dataVersion, explainInfo) =>
+                  implicit val encode = AutomaticJsonEncodeBuilder[ExplainInfo]
+                  Header("X-SODA2-CopyNumber", copyNumber.toString)(resp)
+                  Header("X-SODA2-DataVersion", dataVersion.toString)(resp)
+                  Header("X-SODA2-Secondary-Last-Modified", DateTime.now().toHttpDate)(resp)
+                  resp.setContentType("application/json")
+                  resp.setCharacterEncoding(utf8EncodingName)
+
+                  val writer = new OutputStreamWriter(resp.getOutputStream, StandardCharsets.UTF_8)
+                  JsonUtil.writeJson(writer, Array(explainInfo))
+                  writer.flush()
+                  writer.close()
               }
           }
       }
     }
   }
 
+  trait QueryResultBall
```

Comment:
🎱?

Reply:
I'm bad at names, if you've got a better name for it I'll gladly take it.